### PR TITLE
Bob Hunt 'bomazani'

### PR DESCRIPTION
**Pull request by bomazani**
A, B, C & D are returning expected results.

### CODE REVIEW

File: string1.py [hunk 1]
```diff
@@ -48,8 +53,11 @@ def both_ends(s):
 # Hint: s.replace(stra, strb) returns a version of string s
 # where all instances of stra have been replaced by strb.
 def fix_start(s):
-    """Your code goes here.  Edit this docstring."""
-    return
+    stra = s[0]
+    strb = '*'
```
Comment (madarp) [inline string1.py]: Don't need to declare this .. just use literal '*'

**Comment by madarp** [comment]
Looks terrific!  I can't wait for string2.py!!

**Comment by bomazani** [comment]
Exercises in string2 are completed. I didn't see your note from above until just now. I'll revisit that portion of the code.

File: string2.py [hunk 2]
```diff
@@ -50,14 +61,18 @@ def not_bad(s):
...
 #  a-front + b-front + a-back + b-back
 def front_back(a, b):
-    """Your code goes here.  Edit this docstring."""
-    return
+    back_a = int(math.floor(len(a)/2))
+    front_a = len(a) - back_a
+    back_b = int(math.floor(len(b)/2))
+    front_b = len(b) - back_b
+    c = a[:(front_a)] + b[:(front_b)] + a[(back_a * -1):] + b[(back_b * -1):]
+    return c
```
Comment (madarp) [inline string2.py]: Have a look at the `//` (double slash) operator defined in Python and floor division
https://stackoverflow.com/questions/183853/in-python-2-what-is-the-difference-between-and-when-used-for-division
You don't have to import math for this.

File: string2.py [hunk 3]
```diff
@@ -50,14 +61,18 @@ def not_bad(s):
...
 
 
 # Provided simple test() function used in main() to print
 # what each function returns vs. what it's supposed to return.
 def test(got, expected):
-    """Your code goes here.  Edit this docstring."""
+    """Indicate if individual tests returned expected results"""
```
Comment (madarp) [inline string2.py]: Nice docstring! +2 pts.

**Comment by madarp** [comment]
Closing as completed

**Pull request closed without merging**